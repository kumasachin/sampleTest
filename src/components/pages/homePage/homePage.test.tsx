import React from 'react';
import HomePage from './homePage';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import generateStore from "../../../stateManager/store/generateStore";
import journeyReducers from "../../../stateManager/reducers/journeyReducer";
import { Provider } from "react-redux";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

const mockfn = jest.fn();
describe("homepage", () => {
    let wrapper: any;
    let store: any;
    const props: any = {
      history: { push: mockfn },
      match: { params: {} },
      dataSelectorPrefix: "ccob",
      isDirty: true,
      setSessionTracking: mockfn,
    };
  
    beforeEach(() => {
      store = generateStore(journeyReducers);
      wrapper = mount(
        <Provider store={store}>
          <HomePage {...props} />
        </Provider>
      );
    });
  
    it('renders correctly enzyme', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});