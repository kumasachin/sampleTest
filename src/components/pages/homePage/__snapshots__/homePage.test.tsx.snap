// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage renders correctly enzyme 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <HomePage
    dataSelectorPrefix="ccob"
    history={
      Object {
        "push": [MockFunction],
      }
    }
    isDirty={true}
    match={
      Object {
        "params": Object {},
      }
    }
    setSessionTracking={[MockFunction]}
  />
</Provider>
`;

exports[`renders correctly enzyme 1`] = `<homePage />`;
